src/gemdat/plots/matplotlib/_displacements.py

```diff
@@ -81,19 +81,16 @@ def msd_per_element(*, trajectory: Trajectory) -> plt.Figure:
     fig : matplotlib.figure.Figure
         Output figure
     """
-    grouped = defaultdict(list)
-
-    species = trajectory.species
-
-    for sp, distances in zip(species,
-                             trajectory.distances_from_base_position()):
-        grouped[sp.symbol].append(distances**2)
+    species = list(set(trajectory.species))
 
     fig, ax = plt.subplots()
 
-    for symbol, sq_distances in grouped.items():
-        mean_sq_disp = np.mean(sq_distances, axis=0)
-        ax.plot(mean_sq_disp, lw=0.3, label=symbol)
+    for sp in species:
+        if len(species) > 1:
+            traj = trajectory.filter(sp.symbol)
+        else:
+            traj = trajectory
```

Comment:
I don't understand what this bit does. It loops over the species, but if the lengh of species is bigger then one it filters?

Isn't this essentially the same as this:

```suggestion
    for sp in species:
        traj = trajectory.filter(sp.symbol)
```

Reply:
I was scared about filtering a trajectory multiple times, but it makes no difference at all, you are right!